addons/website_slides/controllers/main.py

```diff
@@ -65,7 +65,7 @@ def _fetch_slide(self, slide_id):
 
     def _set_viewed_slide(self, slide, quiz_attempts_inc=False):
         if request.env.user._is_public() or not slide.website_published or not slide.channel_id.is_member:
-            viewed_slides = request.session.setdefault('viewed_slides', set())
+            viewed_slides = set(request.session.setdefault('viewed_slides', set()))
```

Comment:
In case I can provide a bit of help with the wording, something like this.
```suggestion
            # set(...) ensures backward compatibility for sessions created earlier using a list. todo: remove for v16.1.
            viewed_slides = set(request.session.setdefault('viewed_slides', set()))
```